Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py

```diff
@@ -0,0 +1,229 @@
+        N(d) = (univariate) cumulative standard normal distribution function (from -inf to d)
+        B = face value (principal) of the debt
+        D = dividend + interest payout
+        V = value of firm’s assets
+        σ (sigma) = standard deviation of firm value changes (returns in V)
+        τ (tau)  = time to debt’s maturity
+        µ (mu) = interest rate
+        
+        
+    
+    This alpha is part of the Benchmark Alpha Series created by QuantConnect which are open
+    sourced so the community and client funds can see an example of an alpha.
+'''
+
+
+
+import scipy.stats as sp
+import pandas as pd
+import numpy as np
+from datetime import datetime, timedelta
+
+from Risk.NullRiskManagementModel import NullRiskManagementModel
+from Portfolio.EqualWeightingPortfolioConstructionModel import EqualWeightingPortfolioConstructionModel
+from Execution.ImmediateExecutionModel import ImmediateExecutionModel
+
+
+class ContingentClaimAnalysisDefaultPredictionAlpha(QCAlgorithmFramework):
+
+    def Initialize(self):
+
+        ## Set requested data resolution and variables to help with Universe Selection control
+        self.UniverseSettings.Resolution = Resolution.Daily
+        self.month = None
+        self.symbols = None
+        
+        ## Declare single variable to be passed in multiple places -- prevents issue with conflicting start dates declared in different places
+        self.SetStartDate(2018,1,1)
+        self.SetCash(100000)
+        
+        ## SPDR Small Cap ETF is a better benchmark than the default SP500
+        self.SetBenchmark('IJR')
+
+        ## Set Universe Selection Model
+        self.SetUniverseSelection(FineFundamentalUniverseSelectionModel(self.CoarseSelectionFunction, self.FineSelectionFunction, None, None))
+        
+        ## Set CCA Alpha Model
+        self.SetAlpha(ContingentClaimsAnalysisAlphaModel())
+        
+        ## Set Portfolio Construction Model
+        self.SetPortfolioConstruction(EqualWeightingPortfolioConstructionModel())
+        
+        ## Set Execution Model
+        self.SetExecution(ImmediateExecutionModel())
+        
+        ## Set Risk Management Model
+        self.SetRiskManagement(NullRiskManagementModel())
+
+
+    def CoarseSelectionFunction(self, coarse):
+        ## Boolean controls so that our symbol universe is only updated once per month
+        if self.Time.month == self.month:
+            return self.symbols
+        else:
+            self.month = self.Time.month
+
+            ## Sort by dollar volume, lowest to highest
+            sortedByDollarVolume = sorted(coarse, key=lambda x: x.DollarVolume, reverse=True)
+
+            ## Filter for assets with fundamental data
+            filtered = [ x.Symbol for x in sortedByDollarVolume ]
+        
+            ## Return smallest 750 -- idea is that smaller companies are most likely to go bankrupt than blue-chip companies
+            self.symbols = filtered[:750]
```

Comment:
The `FineFundamentalUniverseSelectionModel` will automatically remove the Coarse Fundamental that `HadFundamentalData` property is false. However, if we want to ensure that 750 symbols arrive at Fine Fundamental selection function, we need to filter in this method (before line 81).

Reply:
Not sure I follow we return all symbols from Coarse and then take 750 after fineFilter has been created?